sdk/communication/Azure.Communication.MediaComposition/src/MediaCompositionClient.cs

```diff
@@ -0,0 +1,314 @@
+        { }
+
+        private MediaCompositionClient(string endpoint, AzureKeyCredential keyCredential, MediaCompositionClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(keyCredential), options)
+        { }
+
+        private MediaCompositionClient(string endpoint, HttpPipeline httpPipeline, MediaCompositionClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RestClient = new MediaCompositionRestClient(_clientDiagnostics, httpPipeline, new Uri(endpoint), options.ApiVersion);
+        }
+
+        #endregion
+
+        /// <summary>Initializes a new instance of <see cref="MediaCompositionClient"/> for mocking.</summary>
+        protected MediaCompositionClient()
+        {
+            _clientDiagnostics = null;
+            RestClient = null;
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id to use. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<MediaCompositionBody>> GetAsync(string mediaCompositionId, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Get");
+            scope.Start();
+            try
+            {
+                return await RestClient.GetAsync(mediaCompositionId, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id to use. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual Response<MediaCompositionBody> Get(string mediaCompositionId, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Get");
+            scope.Start();
+            try
+            {
+                return RestClient.Get(mediaCompositionId, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id. </param>
+        /// <param name="layout"> The layout to compose the media. </param>
+        /// <param name="inputs"> The media inputs. </param>
+        /// <param name="outputs"> The media outputs. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<MediaCompositionBody>> CreateAsync(string mediaCompositionId, MediaCompositionLayout layout, IDictionary<string, MediaInput> inputs, IDictionary<string, MediaOutput> outputs, CancellationToken cancellationToken = default)
```

Comment:
Do we feel that layout, inputs, outputs should be required on creation?
How would the alternative of passing the whole MediaCompositionBody as an argument feel, pro/cons?

Reply:
Yea, great point Dom and definitely something worth discussion.

I've created a sample file to get a feel of how the SDK works. Personally, I felt there was lots of nesting when building the inputs and the layouts. The objects can also get quite large. I think passing the entire MediaCompositionBody might make that even more apparent.